lib/api.ts

```diff
@@ -53,132 +22,159 @@ export interface MemberPublicEditing extends MemberPublic {
-    getBase({ name: "Focuses", view: "Approved" }),
-    getBase({ name: "Industries", view: "Approved" }),
-    getBase({ name: "Regions" }),
-  ]).then((values) => {
-    const [members, focuses, industries, regions] = values;
-    return members
-      .map((member) => {
-        const regionLookup =
-          regions.find((region) => {
-            const memberRegion = member.fields["Region"];
-            if (memberRegion && Array.isArray(memberRegion)) {
-              return region.id === member.fields["Region"][0];
-            }
-          })?.fields["Name"] || null;
+export interface DocumentData {
+  id: string;
+  fields: any;
+}
 
-        const focusLookup = () => {
-          const memberFocus = member.fields["Focus"];
-          if (memberFocus && Array.isArray(memberFocus)) {
-            return memberFocus.map((foc) => {
-              return (
-                focuses
-                  .filter((thisFoc) => foc === thisFoc.id)
-                  .map((foc) => {
-                    return {
-                      name:
-                        typeof foc.fields["Name"] === "string"
-                          ? foc.fields["Name"]
-                          : null,
-                      id:
-                        typeof foc.fields["ID"] === "string"
-                          ? foc.fields["ID"]
-                          : null,
-                    };
-                  })[0] || null
-              );
-            });
-          }
-          return null;
-        };
+export async function getFirebaseTable(table: string) {
+  const documentsCollection = collection(db, table);
+  const documentsSnapshot = await getDocs(documentsCollection);
+  const documentsData = documentsSnapshot.docs.map((doc) => ({
+    id: doc.id,
+    fields: doc.data(),
+  }));
+  return documentsData;
+}
 
-        const industryLookup = () => {
-          const memberIndustry = member.fields["Industry"];
-          if (memberIndustry && Array.isArray(memberIndustry)) {
-            return memberIndustry.map((ind) => {
-              return (
-                industries
-                  .filter((thisInd) => ind === thisInd.id)
-                  .map((ind) => {
-                    return {
-                      name:
-                        typeof ind.fields["Name"] === "string"
-                          ? ind.fields["Name"]
-                          : null,
-                      id:
-                        typeof ind.fields["ID"] === "string"
-                          ? ind.fields["ID"]
-                          : null,
-                    };
-                  })[0] || null
-              );
-            });
-          }
-          return null;
-        };
+function regionLookup(member: DocumentData, regions: DocumentData[]) {
+  return (
+    regions.find((region) => {
+      const memberRegion = member.fields.regions;
+      // TODO: Change this check if we're expecting only one region per member
```

Comment:
Small kine confused by this TODO. If the member region is an array we find a match only for the first region. If memberRegion is not an array, shouldn't we return `region.id === memberRegion.id` or log something?

Reply:
yea this was worded poorly, meant to say we should clarify if regions will continue to be an array

I think @taylorkmho had implemented that with the forward-thinking intention of allowing users to eventually list two places for people who work half-half (e.g. California **and** Hawaii)

I think it makes sense to remove this TODO if that's still something we want to eventually provide

Reply:
removed this TODO in the next commit for now